department-of-veterans-affairs/vets-api
# Burial | Validate attachments

## Summary

*This work is behind a feature toggle (flipper): NO*

When submitting Benefits claims, the Lighthouse::SubmitBenefitsIntakeClaim job doesn't validate attachments before submitting the claim. When the attachments reach Lighthouse, they're rejected by VBADocuments::UploadValidations. The error from VBADocuments is logged to DataDog, but since it occurs after submission, it doesn't trigger an error in Lighthouse::SubmitBenefitsIntakeClaim.

We shouldn't submit claims to Lighthouse with invalid attachments. This PR updates Lighthouse::SubmitBenefitsIntakeClaim to validate attachments when adding them (see PensionBenefitIntakeJob#perform and PensionBenefitIntakeJob#process_document for an example of an existing implementation). This will cause an error to be raised in the Sidekiq job, triggering retries and logging.

Added additional StatsD metrics to Lighthouse::SubmitBenefitsIntakeClaim to track job success and failure, as well as incidents of invalid evidence.

This is a minimal implementation to move the error into our environment to allow us to more closely monitor the problem, not a final fix. 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/92981

## Testing done

     - Run vets-api and vets-website
     - Visit [the Burial form](http://localhost:3001/burials-memorials/veterans-burial-allowance/apply-for-allowance-form-21p-530ez/introduction), and fill out the form. Attach a PDF that is extremely large (over 78in wide), and submit the form. You should see errors in the console. 
     - Visit the Burial form, and fill out the form. Attach a standard valid PDF, and submit the form. The form should submit successfully.

## Screenshots

Valid Attachment
<img width="717" alt="Screenshot 2024-09-18 at 9 51 19 AM" src="https://github.com/user-attachments/assets/09c4533b-f4dd-47c4-9b5e-ef740e3b8a7b">

Invalid Attachment
<img width="718" alt="Screenshot 2024-09-17 at 3 49 19 PM" src="https://github.com/user-attachments/assets/adcc51e8-c901-448f-b345-c8bddab79446">

## What areas of the site does it impact?
Forms using Lighthouse::SubmitBenefitsIntakeClaim job (Burial, VRE, EducationCareerCounseling).

## Acceptance criteria

- [x] The Lighthouse::SubmitBenefitsIntakeClaim job should fail if an attachment is invalid
